src/lightning/pytorch/tuner/lr_finder.py

```diff
@@ -393,7 +393,11 @@ def on_train_batch_end(
         if (trainer.fit_loop.batch_idx + 1) % trainer.accumulate_grad_batches != 0:
             return
 
-        if outputs is None:
+        # _AutomaticOptimization.run turns None STEP_OUTPUT into an empty dict
+        if outputs is None or not outputs:
```

Comment:
This would cover both cases
```suggestion
        if not outputs:
```